examples/lifelong_learning/robot_dog_delivery/RFNet/predict.py

```diff
@@ -0,0 +1,139 @@
+
+
+def init_ll_job(**kwargs):
+    estimator = Model(num_class=31,
+                      weight_path=kwargs.get('weight_path'),
+                      save_predicted_image=True,
+                      merge=True)
+
+    task_allocation = {
+        "method": "TaskAllocationDefault"
+    }
+    unseen_task_allocation = {
+        "method": "UnseenTaskAllocationDefault"
+    }
+
+    unseen_sample_recognition = {
+        "method": "OodIdentification",
+        "param": {
+            "OOD_thresh": float(kwargs.get("OOD_thresh")),
+            "backup_model": kwargs.get("OOD_backup_model"),
+            "OOD_model_path": kwargs.get("OOD_model"),
+            "preprocess_func": preprocess_frames,
+            "base_model": Model
+        }
+    }
+
+    # unseen_sample_recognition = {
+    #     "method": "SampleRegonitionRobotic"
+    # }
+
+    inference_integrate = {
+        "method": "InferenceIntegrateByType"
+    }
+
+    ll_job = LifelongLearning(
+        estimator,
+        unseen_estimator=unseen_task_processing,
+        task_definition=None,
+        task_relationship_discovery=None,
+        task_allocation=task_allocation,
+        task_remodeling=None,
+        inference_integrate=inference_integrate,
+        task_update_decision=None,
+        unseen_task_allocation=unseen_task_allocation,
+        unseen_sample_recognition=unseen_sample_recognition,
+        unseen_sample_re_recognition=None)
+
+    return ll_job
+
+
+def unseen_task_processing():
+    return "Warning: unseen sample detected."
+
+
+def predict():
+    ll_job = init_ll_job()
+
+    camera_address = Context.get_parameters('video_url')
+    # use video streams for testing
+    camera = cv2.VideoCapture(camera_address)
+    fps = 10
+    nframe = 0
+    while 1:
+        ret, input_yuv = camera.read()
+        if not ret:
+            time.sleep(5)
+            camera = cv2.VideoCapture(camera_address)
+            continue
+
+        if nframe % fps:
+            nframe += 1
+            continue
+
+        img_rgb = cv2.cvtColor(input_yuv, cv2.COLOR_BGR2RGB)
+        nframe += 1
+        if nframe % 1000 == 1:  # logs every 1000 frames
+            warnings.warn(f"camera is open, current frame index is {nframe}")
+
+        img_rgb = cv2.resize(np.array(img_rgb), (2048, 1024),
+                             interpolation=cv2.INTER_CUBIC)
+        img_rgb = Image.fromarray(img_rgb)
+        data = {'image': img_rgb, "depth": img_rgb, "label": img_rgb}
+        data = preprocess(data)
+        print(postprocess)
+        print("Inference results:", ll_job.inference(
+            data=data, post_process=postprocess))
+
+
+def predict_batch():
+    ll_job = init_ll_job()
+
+    test_dataset_url = Context.get_parameters("test_dataset_url")
+    test_data = TxtDataParse(data_type="test", func=_load_txt_dataset)
+    test_data.parse(test_dataset_url, use_raw=False)
+
+    return ll_job.inference(data=test_data)
+
+
+if __name__ == '__main__':
+    print(predict())

```

Comment:
The following images under `ramp` directory should be moved to net drive instead of commiting here, if it is really needed.

Reply:
done